README.rst

```diff
@@ -68,6 +68,59 @@ Add the following to any ``ModelAdmin`` with ``date_hierarchy`` to prevent the d
         date_hierarchy_drilldown = False
 
 
+To change the default dates generated by the template tag for any level in the hierarchy, implement a
+function called ``get_date_hierarchy_drilldown(self, year_lookup=None, month_lookup=None)`` on the ``ModelAdmin``.
+The function receive the date hierarchy filter and exected to returns a list of dates to offer for drill-down.
+
+For example, a custom drill-down that offers only past dates:
+
+
+.. code-block:: python
+
+
+    import datetime
+    import calendar
+
+    from django.contrib import admin
+
+    ...
+
+    @admin.register(MyModel)
+    class MyModelAdmin(admin.ModelAdmin):
+        date_hierarchy = 'created'
+        date_hierarchy_drilldown = False
+
+        def get_date_hierarchy_drilldown(self, year_lookup, month_lookup):
+            """Drill-down only on past dates."""
+
+            today = timezone.now().date()
+
+            if year_lookup is None and month_lookup is None:
+                # Past 3 years.
+                return (
+                    datetime.date(y, 1, 1)
+                    for y in range(today.year - 2, today.year + 1)
+                )
+
+            elif year_lookup is not None and month_lookup is None:
+                # Past months of selected year.
+                this_month = today.replace(day=1)
+                return (
+                    month for month in (
+                        datetime.date(int(year_lookup), month, 1)
+                        for month in range(1, 13)
+                    ) if month <= this_month
+                )
+
+            elif year_lookup is not None and month_lookup is not None:
+                # Past days of selected month.
+                return (
+                    day for day in (
+                        datetime.date(year_lookup, month_lookup, 1) + datetime.timedelta(days=i)
+                        for i in range(calendar.monthrange(year_lookup, month_lookup)[1])
```

Comment:
Extract `calendar.monthrange(year_lookup, month_lookup)[1]` to a variable `days_in_month` (I had to check what the thing does.

I think it would be better to avoid `timedelta` and just iterate over `day in range(1, days_in_month+1)` and then `datetime.date(year_lookup, month_lookup, day)`.